pkg/zkcertificate/certificate.go

```diff
@@ -191,8 +197,8 @@ func SignCertificate(
 		return nil, fmt.Errorf("hash message: %w", err)
 	}
 
-	// TODO: Why mod here? It doesn't crash without mod
-	// message = message.Mod(message, utils.NewIntFromString("2736030358979909402780800718157159386076813972158567259200215660948447373040"))
+	// take modulo of the message to get it into the mod field supported by eddsa
+	message = message.Mod(message, utils.NewIntFromString(eddsaPrimeFieldMod))
```

Comment:
I feel like we should extract hash function that would call `poseidon.Hash` and take this modulo, so we won't forget it anywhere else